docs/developer-docs/multi-chain/ethereum/evm-rpc/costs.mdx

```diff
@@ -0,0 +1,136 @@
+
+## Costs
+
+JSON-RPC requests typically cost between 10^8 and 10^9 cycles, which is equivalent to approximately $0.0001 - $0.001 USD.
+
+Because the Candid-RPC methods of the EVM RPC canister use built-in retries and multiple providers, the amount of cycles required for each RPC call isn't predictable beforehand. The idea for these methods is to send a maximum cycles budget to perform the request.
+
+The following formula shows how to calculate the cycles cost for an RPC request:
+
+```
+(
+  3M
+  + 60K * nodes_in_subnet // Number of nodes in the subnet
+  + 400 * request_size // Size of the HTTP request in bytes
+  + 800 * max_response // Maximum HTTP response size in bytes
+  + provider_cost // Fixed cost for selected RPC provider
+) * nodes_in_subnet //  Fixed canister overhead cost per node in subnet
+```
+
+Here is an approximate cost breakdown in USD for an RPC request on the Ethereum mainnet using Cloudflare Web3 and assuming a 13-node subnet:
+
+- Total cost: ~ $0.0021 (assuming consensus between 3 RPC providers)
+
+  - Cost per RPC provider: ~ $0.000686 (assuming 1kb request and 1kb response)
+
+    - Canister overhead: ~ $0.0000364
+
+    - HTTPS outcall overhead: ~ $0.00025
+
+    - JSON Request: ~ $0.0002 / kB
+
+    - JSON Response: ~ $0.0002 / kB
+
+Note that the cost is multiplied by the number of RPC services used for the agreement / consensus logic. If you specify three different services, it will cost three times as much as a call to a single RPC provider.
+
+## Attaching the correct amount of cycles
+
+To determine how many cycles need to be sent with your RPC call, you can use the `requestCost` query method:
+
+```
+`requestCost`
+
+requestCost : (
+  source : JsonRpcSource,
+  jsonRequest : text,
+  maxResponseBytes :  nat64
+) -> (
+  Result<nat, RpcError>
+) query;
+```
+
+This query method accepts the same arguments as the canister's [`request`](evm-rpc-canister.mdx#custom-json-rpc-requests) and returns the number of cycles to send with an equivalent call to the `request` method.
+
+Once you have determined how many cycles your call will need, you can send them in your call through a CDK or using `dfx` with the `--with-cycles` flag.
+
+<AdornedTabs groupId="languages">
+<TabItem value="motoko" label="Motoko" default>
+
+```motoko
+
+import EvmRpc "canister:evm_rpc";
+import Cycles "mo:base/ExperimentalCycles";
+
+Cycles.add<system>(1000000000);
+let result = await EvmRpc.eth_getBlockByNumber(services, null, #Latest);
+```
+
+
+</TabItem>
+<TabItem value="rust" label="Rust">
+
+```rust
+
+use ic_cdk::api::call::call_with_payment128;
+use declarations::evm_rpc::{Block, BlockTag, RpcError, RpcService, EVM_RPC as evm_rpc};
+
+let cycles = 1000000000;
+ic_cdk::api::call::call_with_payment128(
+    evm_rpc.0,
+    "eth_getBlockByNumber",
+    (
+        RpcServices::EthMainnet(None),
+        (),
+        BlockTag::Number(123.into()),
+    ),
+    cycles,
+)
+.await
+.unwrap();
+
+```
```

Comment:
Updated the example to show how to access the result:

````suggestion
```rust
use ic_cdk::api::call::call_with_payment128;
use declarations::evm_rpc::{Block, BlockTag, MultiGetBlockByNumberResult, RpcError, RpcService, EVM_RPC as evm_rpc};

let cycles = 1000000000;
let (results,): (MultiGetBlockByNumberResult,) = call_with_payment128(
    evm_rpc.0,
    "eth_getBlockByNumber",
    (
        RpcServices::EthMainnet(None),
        (),
        BlockTag::Number(19709434.into()),
    ),
    cycles,
)
.await
.unwrap();
```
````